scripts/convert_llama_to_dcp.py

```diff
@@ -0,0 +1,141 @@
+                del shards[i][f"layers.{layer}.ffn_norm.weight"]
+
+            for wn, nh in [
+                ("wq", n_heads_per_shard),
+                ("wk", n_kv_heads_per_shard),
+                ("wv", n_kv_heads_per_shard),
+            ]:
+                state_dict[f"layers.{layer}.attention.{wn}.weight"] = torch.cat(
+                    [
+                        shards[i][f"layers.{layer}.attention.{wn}.weight"].view(
+                            nh, dims_per_head, dim
+                        )
+                        for i in range(len(shards))
+                    ],
+                    dim=0,
+                ).reshape(nh * len(shards) * dims_per_head, dim)
+                for i in range(len(shards)):
+                    del shards[i][f"layers.{layer}.attention.{wn}.weight"]
+
+            state_dict[f"layers.{layer}.attention.wo.weight"] = torch.cat(
+                [
+                    shards[i][f"layers.{layer}.attention.wo.weight"]
+                    for i in range(len(shards))
+                ],
+                dim=1,
+            )
+            for i in range(len(shards)):
+                del shards[i][f"layers.{layer}.attention.wo.weight"]
+
+            state_dict[f"layers.{layer}.feed_forward.w1.weight"] = torch.cat(
+                [
+                    shards[i][f"layers.{layer}.feed_forward.w1.weight"]
+                    for i in range(len(shards))
+                ],
+                dim=0,
+            )
+            for i in range(len(shards)):
+                del shards[i][f"layers.{layer}.feed_forward.w1.weight"]
+
+            state_dict[f"layers.{layer}.feed_forward.w2.weight"] = torch.cat(
+                [
+                    shards[i][f"layers.{layer}.feed_forward.w2.weight"]
+                    for i in range(len(shards))
+                ],
+                dim=1,
+            )
+            for i in range(len(shards)):
+                del shards[i][f"layers.{layer}.feed_forward.w2.weight"]
+
+            state_dict[f"layers.{layer}.feed_forward.w3.weight"] = torch.cat(
+                [
+                    shards[i][f"layers.{layer}.feed_forward.w3.weight"]
+                    for i in range(len(shards))
+                ],
+                dim=0,
+            )
+            for i in range(len(shards)):
+                del shards[i][f"layers.{layer}.feed_forward.w3.weight"]
+
+        state_dict["norm.weight"] = shards[0]["norm.weight"]
+        for i in range(len(shards)):
+            del shards[i]["norm.weight"]
+        state_dict["tok_embeddings.weight"] = torch.cat(
+            [shards[i]["tok_embeddings.weight"] for i in range(len(shards))], dim=0
+        )
+        for i in range(len(shards)):
+            del shards[i]["tok_embeddings.weight"]
+        state_dict["output.weight"] = torch.cat(
+            [shards[i]["output.weight"] for i in range(len(shards))], dim=0
+        )
+        for i in range(len(shards)):
+            del shards[i]["output.weight"]
+
```

Comment:
Pre-compute the `freqs_cis` key.

```suggestion
    # NOTE: We precompute freqs_cis because they cannot be computed when loading the checkpoint in TorchTitan.
    state_dict["freqs_cis"] = precompute_freqs_cis(
        dims_per_head,
        max_seq_len,
        params.get("rope_theta", 500000),
    )
```